libs/s25main/Cheats.h

```diff
@@ -0,0 +1,130 @@
+     */
+    bool isCheatModeOn() const { return isCheatModeOn_; }
+
+    // Classic S2 cheats
+
+    /** The classic F7 cheat.
+     * Does not modify game state, merely tricks clients into revealing the whole map.
+     * In the background, visibility is tracked as expected, i.e. if you reveal the map, send a scout and unreveal the
+     * map, you will see what was scouted.
+     */
+    void toggleAllVisible();
+    bool isAllVisible() const { return isAllVisible_; }
+
+    /** The classic build headquarters cheat.
+     * This function is used to check if the cheat building can be placed at the chosen point when opening the activity
+     * window.
+     *
+     * @param mp - The map point where the user clicked to open the activity window.
+     * @return true if the building can be placed, false otherwise
+     */
+    bool canPlaceCheatBuilding(const MapPoint& mp) const;
+    /** The classic build headquarters cheat.
+     * This function is used to place the cheat building at the chosen point.
+     * The building is immediately fully built, there is no need for a building site.
+     *
+     * @param mp - The map point at which to place the building.
+     * @param player - The player to whom the building should belong.
+     */
+    void placeCheatBuilding(const MapPoint& mp, const GamePlayer& player);
+
+    /** The classic ALT+1 through ALT+6 cheat which changes the game speed.
+     *
+     * @param speedIndex - 0 is normal, 1 is faster, 2 is even faster, etc.
+     */
+    void setGameSpeed(uint8_t speedIndex);
+
+    // RTTR cheats
+
+    /** Shares control of the (human) user's country with the AI. Both the user and the AI retain full control of the
+     * country, so the user can observe what the AI does or "cooperate" with it.
+     */
+    void toggleHumanAIPlayer();
+
+    void armageddon();
+
+    enum class ResourceRevealMode
+    {
+        Nothing,
+        Ores,
+        Fish,
+        Water
+    };
+    /** Tells clients which resources to reveal:
+     * Nothing - reveal nothing
+     * Ores - reveal ores
+     * Fish - reveal ores and fish
+     * Water - reveal ores, fish and water
+     */
+    ResourceRevealMode getResourceRevealMode() const;
+    void toggleResourceRevealMode();
+
+    using PlayerIDSet = std::unordered_set<unsigned>;
+    /** Destroys all buildings of given players, effectively defeating them.
+     *
+     * @param playerIds - Set of IDs of players.
+     */
+    void destroyBuildings(const PlayerIDSet& playerIds);
+    /** Destroys all buildings of AI players.
+     */
+    void destroyAllAIBuildings();
+
+private:
+    /** Checks if cheats can be turned on at all.
+     *
+     * @return true if if cheats can be turned on, false otherwise
+     */
+    bool canCheatModeBeOn() const;
+
+    std::unique_ptr<CheatCommandTracker> cheatCmdTracker_;
+    bool isCheatModeOn_ = false;
+    bool isAllVisible_ = false;
+    GameWorldBase& world_;
```

Comment:
we should use a `std::reference_wrapper` instead of a plain reference

Reply:
I don't quite see what the benefit of `reference_wrapper` would be here? `world_` is literally meant as a run of the mill C++ reference. Also there is just one use of `reference_wrapper` anywhere in RTTR code (excluding turtle) and it's in s25util where it is inside a `std::optional`, where a plain reference cannot be.